File: 2766-relocate-marbles/2766-relocate-marbles.java
class Solution {
    public List<Integer> relocateMarbles(int[] nums, int[] moveFrom, int[] moveTo) {
                final int n = moveFrom.length;
        final Set<Integer> set = new HashSet<>();

        for(final int num : nums)
            set.add(num);

        for(int i = 0; i < n; ++i) {
            set.remove(moveFrom[i]);
            set.add(moveTo[i]);
        }

        final List<Integer> result = new ArrayList<>(set);

        Collections.sort(result);

        return result;
    }
}

File: 2766-relocate-marbles/2766-relocate-marbles.py
class Solution:
    def relocateMarbles(self, nums: List[int], moveFrom: List[int], moveTo: List[int]) -> List[int]:
        nums = set(nums)
        for from_pos, to_pos in zip(moveFrom, moveTo):
            nums.discard(from_pos)  # Remove the marble from current position
            nums.add(to_pos)         # Add the marble to new position
        return sorted(nums)        